src/middlewared/middlewared/plugins/docker/state_setup.py

```diff
@@ -0,0 +1,69 @@
+import os
+import shutil
+import uuid
+
+from datetime import datetime
+
+from middlewared.service import private, Service
+
+from .state_utils import DATASET_DEFAULTS, docker_datasets, docker_dataset_custom_props, docker_dataset_update_props
+from .utils import applications_ds_name
+
+
+class DockerSetupService(Service):
+
+    class Config:
+        namespace = 'docker.setup'
+        private = True
+
+    @private
+    async def status_change(self):
+        config = await self.middleware.call('docker.config')
+        if not config['pool']:
+            return
+
+        await self.pools(config)
+
+    @private
+    async def pools(self, config):
+        return await self.create_update_docker_datasets(applications_ds_name(config['pool']))
+
+    @private
+    async def create_update_docker_datasets(self, docker_ds):
```

Comment:
Are we sure we need to do this? We're iterating in a for loop and running `zfs get -o blah` for each dataset.

Reply:
This code will not be used as we have changed how we want to handle actual datasets reality on the filesystem, so this will be addressed in my subsequent PR

Reply:
Fantastic!